team-4-new-graph

waynewangyuxuan opened the pull request:
Commits Merged.
Resulting graphs are saved in the folder /steps.

digit-google committed:
Add InputsCollector class.

Add a new class to wrap the logic needed to implement the
`inputs` tool correctly (see Issue #2482 for details), and
provide a unit-test for it.

digit-google committed:
Fix `inputs` tool implementation.

This uses the InputsCollector class introduced in the
previous patch to implement the tool properly. Results
are still shell-escaped and sorted alphabetically.

Fixed #2482

digit-google committed:
Add new options to `inputs` tool.

Add new options to the `inputs` tool in order to change
the format of its output:

- `--no-shell-escape` to avoid shell-escaping the results.

- `--dependency-order` to return results in dependency order,
  instead of sorting them alphabetically.

- `--print0` to use \0 as the list separator in the list,
  useful to process the target paths with `xargs -0` and
  similar tools.

YuWei-CH committed:
Create a graph for HelloWorld directly on memory. Updated According to
Previous peer review contents.

YuWei-CH committed:
init

YuWei-CH committed:
init

CircleCI Commit

Update CI

update README.md

add sudo

Updated CMakeLists.txt to include clang-tidy configuration

change clang test name

update clangtest file

update clang-tidy/circleCI use. Learned from team5's implementation

update yml file

typos

init

CircleCI Commit

Update CI

update README.md

add sudo

test folder wrong

YuWei-CH committed:
init

CircleCI Commit

Update CI

update README.md

add sudo

YuWei-CH committed:
CircleCI Commit

YuWei-CH committed:
Update CI

YuWei-CH committed:
update README.md

Commit:
Updated CMakeLists.txt to include clang-tidy configuration

Commit:
Updated CMakeLists.txt to include clang-tidy configuration

change clang test name

update clangtest file

YuWei-CH committed:
add sudo

Commit:
change clang test name

Commit:
update clangtest file

YuWei-CH committed:
Merge pull request #5 from YuWei-CH/yuwei-dev

Update CI && README.md

YuWei-CH committed:
generate hello_world graph using ninja

reorganize

fix: fix createHelloWorldGraph using ninja codes

add steps

waynewangyuxuan committed:
Merge pull request #6 from YuWei-CH/wayne

clang-tidy

Frankbz committed:
fix: fix createHelloWorldGraph using ninja codes

kyotov commented on CMakeLists.txt:
```diff
@@ -311,7 +311,28 @@ if(BUILD_TESTING)
+  endif()
+  endif()
+# Define the Ninja test target
+add_test(NAME NinjaTest COMMAND ninja_test)
```
there are more tests than this

kyotov commented on steps/hello-world-graph/test1.cc:
```diff
@@ -0,0 +1,55 @@
+#include "state.h"
+#include "util.h"
+
+void CreateHelloWorldGraph(State* state) {
```
this is pretty good!@

YuWei-CH committed:
Merge branch 'master' of github.com:YuWei-CH/ShadowDash

Yaxuan-w commented on CMakeLists.txt:
```diff
@@ -84,6 +84,45 @@ else()
 target_include_directories(libninja-re2c PRIVATE src)
 
+# --- Create shared library ---
+add_library(libninja_shared SHARED
+	src/build_log.cc
+	src/build.cc
+	src/clean.cc
+	src/clparser.cc
+	src/dyndep.cc
+	src/dyndep_parser.cc
+	src/debug_flags.cc
+	src/deps_log.cc
+	src/disk_interface.cc
+	src/edit_distance.cc
+	src/elide_middle.cc
+	src/eval_env.cc
+	src/graph.cc
+	src/graphviz.cc
+	src/json.cc
+	src/line_printer.cc
+	src/manifest_parser.cc
+	src/metrics.cc
+	src/missing_deps.cc
+	src/parser.cc
+	src/state.cc
+	src/status_printer.cc
+	src/string_piece_util.cc
+	src/util.cc
+	src/version.cc
+)
+
+# Link re2c library if required
+target_link_libraries(libninja_shared PRIVATE libninja-re2c)
+
+# Include directories for the shared library
+target_include_directories(libninja_shared PUBLIC ${CMAKE_SOURCE_DIR}/src)
+
+# Set shared library properties
```
Could you please add more detailed explanation about the functionality of those code in comment..?

YuWei-CH replied:
Thanks for your comments. The purpose of the code above is to generate a .so with the ninja files we need during the build Ninja process. We later realized that this is not necessary, so we will remove it in the next commits.

kyotov replied:
yes, remove

Yaxuan-w commented on commands.md:
```diff
@@ -0,0 +1,23 @@
+cd build
+cmake -G "Ninja" -DLLVM_ENABLE_PROJECTS="clang;clang-tools-extra" -DCMAKE_BUILD_TYPE=Release ../llvm
+ninja clang-tidy
+export PATH=$PATH:/home/yuwei/Documents/llvm-project/build/bin
```
It should be better to use a more general `PATH` instead of a personal one?

YuWei-CH replied:
Yes, thank you for pointing that out.

Yaxuan-w commented on commands.md:
It seems like a setup script but the file is in markdown format. Is there any specific reason for this?

waynewangyuxuan replied:
we were using this as an internall readme honestly...but you are right, this is not very tidy

Yaxuan-w commented on src/manifest_parser.h:
```diff
@@ -51,7 +51,7 @@ struct ManifestParser : public Parser {
 
   /// Parse various statement types.
-  bool ParsePool(std::string* err);
+  bool ParsePool(std::string* err); // wokers 
```
What's the meaning of `workers`?

Yaxuan-w commented on src/test1.cc:
```diff
@@ -0,0 +1,127 @@
+#include "state.h"
+#include "util.h"
+using namespace std;
+
```
That would be great to have a more high-level comment that describing the functionality of this file

waynewangyuxuan replied:
definitely, nice call!

Yaxuan-w commented on src/test1.cc:
```diff
@@ -0,0 +1,127 @@
+
+  cout << "Build completed successfully!" << std::endl;
+  return 0;
+  return 0;
```
Is there any reason to have two `return` here?

Yaxuan-w commented on steps/hello-world-graph/commands.md:
1. Seems like a script, why markdown here?
2. Would be great to replace all absolute personal path. i.e.: `/home/yuwei/Documents/ShadowDash/build`

waynewangyuxuan replied:
because it is what yuwei used for testing at the time. The  personal path is there because it is command for this linux system, whereas the rest of us, if I am not mistakenly, has one person on MacOS and one on linux. 

This is not designed to be directly runnable, but more for a guidance for the rest of us to test the intermediate work at the time.

Yaxuan-w commented on steps/hello-world-graph/test1.cc:
1. What's the purpose of this file?
2. What's difference with `src/test1.cc` and they have the same name. That would be great to have a high level description and use a more descriptive file name here.

waynewangyuxuan replied:
you are absolutely right. This is test1.cc is essentially what we are working with.  
the src one should be disregarded.

Yaxuan-w commented on test1:
The binary is not necessary..?

waynewangyuxuan replied:
which binary are you referring to?

YuWei-CH replied:
Yes, this is an executable file for generating new-graph. And we will add our internal used command.md and all binary files to the .gitignore. Thank you so much for pointing out those issues.

Yaxuan-w reviewed (commented):
Collaborate with Qianxi Chen @qianxichen233

Great job! Tests tested if the feature has been implemented successfully.

Few minor suggestions:
1. Directory is a bit messy, would be great to have a README.md for high-level description or add it in description section of this PR
2. Duplicated file name and no instructions about their usage. It would be better to use more descriptive file names / detailed comment at beginning of the file / add READMEs 
3. Lots of commits, would be great to do squash merge here

YuWei-CH committed:
resolve issues for peer code review

waynewangyuxuan committed:
add test result display for circleCI dashboard

Added ctest and performance test.

Try to show test on CI/CD dashboard, followed the instruction, however
we could not get it working somehow.

YuWei-CH committed:
Compile hello_world

YuWei-CH committed:
add a script for cross-platform compilation

kyotov reviewed (changes requested):
This suffers from some of the common problems I have noticed before.

* How is each file in your PR connected to the task at hand?  
  * You should not have files that are irrelevant to the task.  
* Make sure files that should be .gitignore are in .gitignore  
  * E.g. .DS\_Store  
* Make sure secrets are not checked in\!  
* Make sure there is a clear and separated API\!

Not all of these apply to you but some do.
Please address them and let me know and I will look again.

Yaxuan-w commented:
Collaborate with Qianxi Chen @qianxichen233:
Do you guys have any update commits after professor's comment? That would be great if you can point us to the new place, thanks!

waynewangyuxuan commented:
> Collaborate with Qianxi Chen @qianxichen233: Do you guys have any update commits after professor's comment? That would be great if you can point us to the new place, thanks!

Hi, no updates since. We were not sure what the professor meant by the comments, hoping to get a clearer clue during class today and we will make updates ASAP long as we know what he meant. Thanks for asking!

YuWei-CH committed:
Merge branch 'master' of github.com:YuWei-CH/ShadowDash

YuWei-CH committed:
Merge branch 'new-graph' of github.com:YuWei-CH/ShadowDash into new-graph

YuWei-CH committed:
Merge branch 'master' into new-graph

YuWei-CH committed:
update CICD

YuWei-CH committed:
rename

YuWei-CH committed:
update cmakelist

waynewangyuxuan committed:
updated readme
